python_modules/dagster/dagster/_check/__init__.py

```diff
@@ -1645,6 +1670,21 @@ def _element_check_error(
     )
 
 
+def _param_value_mismatch_exception(
+    obj: object,
+    values: Sequence[object],
+    param_name: str,
+    additional_message: Optional[str] = None,
+    allow_none: bool = False,
+) -> ParameterCheckError:
+    allow_none_clause = " or None" if allow_none else ""
+    additional_message = " " + additional_message if additional_message else ""
```

Comment:
not a huge fan of this sort of boolean param. Easy to screw up and non-intuitive

Reply:
It follows an existing pattern in `check` (see `_param_class_mismatch_exception` and IMO is intuitive in the limited context in which this method is used.